src/main/java/com/auth0/jwk/JwkSet.java

```diff
@@ -0,0 +1,78 @@
+package com.auth0.jwk;
+
+import java.util.Collections;
+import java.util.List;
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Represents a set of JSON Web Keys (JWK) used to verify the signature of JWTs.
+ */
+public class JwkSet {
+
+    private final List<Jwk> keys;
+
+    /**
+     * Create a new empty JWK set.
+     */
+    public JwkSet() {
+        this(Collections.<Jwk>emptyList());
+    }
+
+    /**
+     * Creates a new JWK set with a single key.
+     * @param key the JWK. Must not be {@code null}.
+     */
+    public JwkSet(final Jwk key) {
+        this(Collections.singletonList(key));
+        checkArgument(key != null, "The JWK must not be null");
+    }
+
+    /**
+     * Creates a new JWK set with a list of keys.
+     * @param keys the list of keys. Must not be {@code null}.
+     */
+    public JwkSet(final List<Jwk> keys) {
```

Comment:
Consider a Set. If the `Jwk` implements equals and checks against the `kid` (which is optional, btw) we could filter repeated keys. (which shouldn't happen but well)

Reply:
`Jwk` does not implement `equals` or `hashCode`. If we would like this object to be a true Set, we could look at updating `Jwk` to support hashing. One other thing to consider is that the public `getAll` method of `UrlJwkProvider` returns a List; we can create a Set from that List, but that would introduce an inconsistency (`getAll()` returns a List, but `JwkSet` would have a Set).

Reply:
Ah, you're right :(. Keep the List for now.